pkg/virt-controller/watch/snapshot/snapshot.go

```diff
@@ -686,24 +688,77 @@ func (ctrl *VMSnapshotController) getVolumeStorageClassForVolume(namespace strin
 		return ctrl.getVolumeSnapshotClass(*pvc.Spec.StorageClassName)
 	}
 	if volume.VolumeSource.DataVolume != nil {
-		pvcKey := cacheKeyFunc(namespace, volume.VolumeSource.DataVolume.Name)
-		// TODO Change when PVC rename PR is implemented
-
-		obj, exists, err := ctrl.PVCInformer.GetStore().GetByKey(pvcKey)
+		storageClassName, err := ctrl.getStorageClassNameForDV(namespace, volume.VolumeSource.DataVolume.Name)
 		if err != nil {
 			return "", err
 		}
 
-		if !exists {
-			log.Log.V(3).Infof("PVC not in cache [%s]", pvcKey)
-			return "", fmt.Errorf("PVC for the DataVolume not found")
+		return ctrl.getVolumeSnapshotClass(storageClassName)
+	}
+
+	return "", fmt.Errorf("Volume type does not suport snapshots")
+}
+
+func (ctrl *VMSnapshotController) getStorageClassNameForDV(namespace string, dvName string) (string, error) {
+	// First, look up DV's StorageClass
+	key := cacheKeyFunc(namespace, dvName)
+
+	obj, exists, err := ctrl.DVInformer.GetStore().GetByKey(key)
+	if err != nil {
+		return "", err
+	}
+
+	if !exists {
+		log.Log.V(3).Infof("DV not in cache [%s]", key)
+		return "", fmt.Errorf("DV '%s' not found", key)
+	}
+
+	dv := obj.(*cdiv1.DataVolume)
+	if dv.Spec.PVC != nil && *dv.Spec.PVC.StorageClassName != "" {
+		return *dv.Spec.PVC.StorageClassName, nil
+	}
+
+	// Second, see if DV is owned by a VM, and if so, if the DVTemplate has a StorageClass
+	for _, or := range dv.OwnerReferences {
+		if or.Kind == "VirtualMachine" {
+
+			vmKey := cacheKeyFunc(namespace, or.Name)
+			storeObj, exists, err := ctrl.VMInformer.GetStore().GetByKey(vmKey)
+			if err != nil || !exists {
+				continue
+			}
+
+			vm, ok := storeObj.(*kubevirtv1.VirtualMachine)
+			if !ok {
+				continue
+			}
+
+			for _, dvTemplate := range vm.Spec.DataVolumeTemplates {
+				if dvTemplate.Name == dvName && dvTemplate.Spec.PVC != nil {
+					return *dvTemplate.Spec.PVC.StorageClassName, nil
```

Comment:
It is legit for `dvTemplate.Spec.PVC.StorageClassName` to be nil

Reply:
OK